mantidproject/mantid
# Export dialog in DrILL

**Description of work.**

This PR adds an automatic export system to the DrILL interface. When the reduction of a sample is done, the automatic export is triggered and exported files are written to the default save directory.

Each acquisition mode has a number of export algorithm associated to it. Some of them are activated by default, some are not. The activated algorithms can be seen and changed through a dedicated dialog in DrILL interface.

**To test:**

Export dialog:
* open the DrILL interface
* open the export dialog (mdi-application-export icon in the tool bar - https://pictogrammers.github.io/@mdi/font/3.6.95/)
* observe the selected export algorithm

Export:
With data in `SystemTest/ILL/D11/`:
* check your default save directory
* open the DrILL interface, select D11 instrument
* put 2888 in the column `SampleRuns` of the first row
* process that row
* check in your default save directory that the automatic export produced the right files
* change the selected algorithm in the export dialog, process again and observe the files in your save directory

Part of #29570

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
